doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -94,42 +134,110 @@ Here we define a common platform API to wait for this event in class `SfpUtilBas
+A thread will be started to periodically refresh the DOM sensor information.
 
-### 1.4 Xcvrd daemon flow ###
+In the main loop of the Xcvrd task, it periodically check the integrity the DB, if some SFP info missing, will be added back. 
 
-Xcvrd will spawn a thread to wait for the SFP plug in/out event, when event received, it will update the DB entries accordingly.
+Detailed flow as showed in below chart: 
 
-A timer will be started to periodically refresh the DOM sensor information . 
+![](https://github.com/keboliu/SONiC/blob/master/images/xcvrd-flow.svg)
 
-Detailed flow as showed in below chart: 
+#### 1.4.1 State machine of sfp\_state\_update\_task process ####
+
+In the process of handling SFP change event, a state machine is defined to handle events(including SFP change events) reported from platform level.
 
-![](https://github.com/Azure/SONiC/blob/master/images/transceiver_monitoring_hld/xcvrd_flow.svg)
+        states definition
+          - Initial state: INIT, before receive system ready or a normal event
+          - Final state: EXIT
+          - Other state: NORMAL, after received system-ready or a normal event
+        
+        events definition
+          - SYSTEM_NOT_READY
+          - SYSTEM_BECOME_READY
+          - NORMAL_EVENT
+            - sfp insertion/removal event
+            - sfp error event
+            - timeout returned by sfputil.get_change_event with status = true
+          - SYSTEM_FAIL
+
+        state transition
+          State           event               next state
+          INIT            SYSTEM NOT READY    INIT / EXIT
+          INIT            SYSTEM FAIL         INIT / EXIT
+          INIT            SYSTEM BECOME READY NORMAL
+          NORMAL          SYSTEM BECOME READY NORMAL
+          NORMAL          SYSTEM FAIL         INIT
+          INIT/NORMAL     NORMAL EVENT        NORMAL
+          NORMAL          SYSTEM NOT READY    INIT
+          EXIT            -
+
+
+![](https://github.com/keboliu/SONiC/blob/master/images/xcvrd_state_mahine.svg)
+
+#### 1.4.2 Transceiver error events handling procedure ####
+
+When error events(defined in section 1.3.1) received from some transceiver, the related interface will be added to the TRANSCEIVER_ERROR table, and DOM information will be removed from the DB. 
+
+Before the DOM update thread update the DOM info, it will check the error table first, DOM info updating will be skipped if some port is in the error table. In the Xcvrd main task recovering missing interface info, same check will also be applied.
+
+Currently no explicit "error clear event" is defined, a plug in event will be considered as port recovered from error(on Mellanox platform it does send out a plug in event when recovered from error). 
+
+An explicit "error clear event" can be added if some vendor's platform do have this kind of event.
+
+On transceiver plug in or plug out events, the port will be removed from the error table.     

```

Comment:
>removed from the error table [](start = 60, length = 28)

Could you leave the 'plug in' or 'plug out' status in the table, and change the table name to 'Transceiver Status Table'

Reply:
can do that, updated accordingly.